tez-plugins/tez-protobuf-history-plugin/src/main/java/org/apache/tez/dag/history/logging/proto/DatePartitionedLogger.java

```diff
@@ -71,10 +72,19 @@ public DatePartitionedLogger(Parser<T> parser, Path baseDir, Configuration conf,
 
   private void createDirIfNotExists(Path path) throws IOException {
     FileSystem fileSystem = path.getFileSystem(conf);
+    FileStatus fileStatus = null;
     try {
-      if (!fileSystem.exists(path)) {
+      fileStatus = fileSystem.getFileStatus(path);
+    } catch (FileNotFoundException fnf) {
```

Comment:
We need to check null as well, the exists logic in Hadoop does check for Null + FNF
https://github.com/apache/hadoop/blob/a00b1c06f32564e9e723202d153698affff4eb53/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/FileSystem.java#L1862

I don't think we can pull in the logic of mkdir inside fnf..

The logic is already inside ``createDirIfNotExists`` method,

Reply:
okay, createDirIfNotExists is small enough to be considered a separate method for this logic
agree on the other points as well, however, this null vs. FNF is a bit vague for first sight, so instead of commenting only "ignore" let's clarify our point here, like:
```
} catch (FileNotFoundException fnf) {
   // ignore: regardless of the outcome of FileSystem.getFileStatus call (exception or returning null), we handle the fileStatus == null case later on, so it's safe to ignore this exception now
}
```
does this make sene?

Reply:
yep, updated